libs/wingsdk/src/target-tf-aws/api.ts

```
@@ -0,0 +1,157 @@
+    }
+    const _tf_function: Function = fn;
+
+    new ApiGatewayIntegration(this, `integration-${hash}`, {
+      restApiId: this.api.id,
+      resourceId: apiResource.id,
+      httpMethod: apiMethod.httpMethod,
+      integrationHttpMethod: "POST",
+      type: "AWS_PROXY",
+      uri: _tf_function.invokeArn,
+    });
+
+    const region = new DataAwsRegion(this, "region");
+    const account = new DataAwsCallerIdentity(this, "account");
+
+    new LambdaPermission(this, `permission-${hash}`, {
+      statementId: "AllowExecutionFromAPIGateway",
+      action: "lambda:InvokeFunction",
+      functionName: _tf_function._functionName,
+      principal: "apigateway.amazonaws.com",
+      sourceArn: `arn:aws:execute-api:${region.name}:${account.accountId}:${this.api.id}/*/${apiMethod.httpMethod}${apiResource.path}`,
+    });
+
+    new LambdaPermission(this, "apigw-lambda", {
+      functionName: _tf_function._functionName,
+      action: "lambda:InvokeFunction",
+      principal: "apigateway.amazonaws.com",
+      sourceArn: `${this.api.executionArn}/*/*`,
+    });
+
+    const deployment = new ApiGatewayDeployment(this, "deployment", {
+      restApiId: this.api.id,
+      lifecycle: {
+        createBeforeDestroy: true,
+      },
+    });
+    const stage = new ApiGatewayStage(this, "stage", {
+      restApiId: this.api.id,
+      stageName: "example",
+      deploymentId: deployment.id,
+    });
+    new ApiGatewayMethodSettings(this, "method-settings", {
+      restApiId: this.api.id,
+      stageName: stage.stageName,
+      settings: {},
+      methodPath: "*/*",
+    });
+
+    core.Resource.addConnection({
+      from: this,
+      to: fn,
+      relationship: "on_get_request",
+    });
+  }
+
+  /** @internal */
+  public _bind(host: core.IInflightHost, ops: string[]): void {
+    if (!(host instanceof Function)) {
+      throw new Error("queues can only be bound by tfaws.Function for now");
+    }
+
+    super._bind(host, ops);
```

This bind method is only needed when people try to use a `Api` resource inside a Wing [inflight](https://docs.winglang.io/concepts/inflights). For this first PR we don't need to support that (it's a secondary use case) so I think we can just throw an error here:
```suggestion
    throw new Error("Method not implemented.");
```